visualization/motion_planning_rviz_plugin/src/motion_planning_frame_planning.cpp

```diff
@@ -129,16 +137,48 @@ void MotionPlanningFrame::computePlanButtonClicked()
 void MotionPlanningFrame::computeExecuteButtonClicked()
 {
   if (move_group_ && current_plan_)
-    move_group_->execute(*current_plan_);
+  {
+    // ui_->stop_button->setEnabled(true); // enable stopping
+    bool success = move_group_->execute(*current_plan_);
+    ui_->result_label->setText(success ? "Executed" : "Execution failed");
+    ui_->stop_button->setEnabled(false); // disable stopping
+  }
+
+  // Hm. The current state is only updated after a while (only when using /execute/ service)
+  // TODO: Wait for update to planning scene? Related to #671
+  sleep(1);
+  updateStartStateToCurrent();
+}
+
+void MotionPlanningFrame::computeStopButtonClicked()
+{
+  if (move_group_)
+  {
+    move_group_->stop();
+    ui_->result_label->setText("Stopped");
+  }
 }
 
 void MotionPlanningFrame::computePlanAndExecuteButtonClicked()
 {
   if (!move_group_)
     return;
   configureForPlanning();
-  move_group_->move();
+  ui_->stop_button->setEnabled(true);
+  bool success = move_group_->move();
+  if (success) ui_->result_label->setText("Executed");
+  else ui_->result_label->setText(!ui_->stop_button->isEnabled() ? "Stopped" : "Failed");
   ui_->plan_and_execute_button->setEnabled(true);
+  ui_->stop_button->setEnabled(false);
+  updateStartStateToCurrent();
```

Comment:
Using the move_group action server, the issue wasn't that dramatic, because intermediate updates come through.
However, you are right, in principle the delayed update issue is observable here too. Should be fixed in #716 - hope a lot of people will help on this ;-)